sources/packages/backend/apps/workers/src/services/application/application.service.ts

```diff
@@ -100,4 +106,32 @@ export class ApplicationService extends RecordDataModelService<Application> {
       },
     });
   }
+
+  /**
+   * Updates the application edit status to desired status if not updated yet.
+   * @param applicationId application to be updated.
+   * @param fromStatus expected status of the application.
+   * @param toStatus desired status to be updated.
+   * @returns update result.
+   */
+  async updateApplicationEditStatus(
+    applicationId: number,
+    fromStatus: ApplicationEditStatus,
+    toStatus: ApplicationEditStatus,
+  ): Promise<UpdateResult> {
+    const now = new Date();
+    return this.repo.update(
+      {
+        id: applicationId,
+        applicationEditStatus: fromStatus,
+      },
+      {
+        applicationEditStatus: toStatus,
+        applicationEditStatusUpdatedOn: now,
+        applicationEditStatusUpdatedBy: this.systemUsersService.systemUser,
+        modifier: this.systemUsersService.systemUser,
```

Comment:
Minor. We can assign a `this.systemUsersService.systemUser` to a const and use in both places.

Reply:
Considering `.systemUser` equivalent to a method, I mentioned that. Still not a blocker.